File: web/studio/ASC.Web.Studio/addons/mail/js/helpcenter/page.js
/* 
 * 
 * (c) Copyright Ascensio System Limited 2010-2014
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * http://www.gnu.org/licenses/agpl.html 
 * 
 */

window.helpPage = (function($) {
    var 
        isInit = false,
        _page;

    var init = function() {
        if (isInit === false) {
            isInit = true;
            _page = $('#helpPanel');
        }
    };

    var show = function(helpId) {
        var params = { helpId: helpId };

        if (!_page.text().trim().length) {
            params.update = true;
            serviceManager.getHelpCenterHtml(params, {success: onGetHelpCenterHtml }, ASC.Resources.Master.Resource.LoadingProcessing);
        } else {
            onGetHelpCenterHtml(params, null);
        }
    };

    var onGetHelpCenterHtml = function(params, html)
    {
        if (params.update) {
            _page.html(html);
            messagePage.initImageZoom();
        }

        _page.show();

        showHelpPage(params.helpId);
        mailBox.hideLoadingMask();
    };

    var hide = function() {
        _page.hide();
    };

    return {
        init: init,
        show: show,
        hide: hide
    };

})(jQuery);

File: web/studio/ASC.Web.Studio/Products/Community/Modules/Wiki/Code/Page.cs
/* 
 * 
 * (c) Copyright Ascensio System Limited 2010-2014
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * http://www.gnu.org/licenses/agpl.html 
 * 
 */

using System;
using ASC.Core.Tenants;

namespace ASC.Web.UserControls.Wiki.Data
{
    public class Page : IVersioned
    {
        public int Tenant { get; set; }
        public Guid UserID { get; set; }
        public int Version { get; set; }
        public DateTime Date { get; set; }

        public Guid OwnerID { get; set; }
        public object GetObjectId()
        {
            return PageName;
        }

        public string PageName { get; set; }
        public string Body { get; set; }
        

        public Page()
        {
            PageName = Body = string.Empty;
            Date = TenantUtil.DateTimeNow();
        }      
    }
}